### PR TITLE
ddl: add more tests for multi-schema change

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref #41602

Problem Summary:

#47135 have little functional tests.

### What is changed and how it works?

- Add more functional tests for multi-schema change.
- Fix a small bug that `task_key` is the same among different sub-jobs.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tests/realtikvtest/addindextest/global_sort_test.go [hunk 1]
```diff
@@ -108,3 +108,63 @@ func TestGlobalSortCleanupCloudFiles(t *testing.T) {
...
+	store := realtikvtest.CreateMockStoreAndSetup(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("drop database if exists addindexlit;")
+	tk.MustExec("create database addindexlit;")
+	tk.MustExec("use addindexlit;")
+
+	tk.MustExec("create table t_rowid (a int, b bigint, c varchar(255));")
+	tk.MustExec("create table t_int_handle (a bigint primary key, b varchar(255));")
+	tk.MustExec("create table t_common_handle (a int, b bigint, c varchar(255), primary key (a, c) clustered);")
+	// tk.MustExec(`create table t_partition (a bigint primary key, b int, c char(10)) partition by hash(a) partitions 4;`)
```
Comment (ywqzzy) [inline tests/realtikvtest/addindextest/global_sort_test.go:134]: Why comment out partition table?

Comment (tangenta) [reply tests/realtikvtest/addindextest/global_sort_test.go:134]: Because it takes too long to complete... 4 partitions means 4x import times.

**Review by ywqzzy — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Comment by BornChanger** [comment]
Is it needed in 7.4? @Leavrth

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**